lightning/src/util/ser.rs

```diff
@@ -913,6 +915,82 @@ impl Readable for String {
 	}
 }
 
+/// Represents a hostname with validation for serialization purposes.
+/// The hostname follows the preferred form summarized in e.g. RFC 3696.
+/// Its length is guaranteed to be representable by a single byte.
+/// This serialization is used by Bolt 7 hostnames.
+#[derive(Clone, Debug, PartialEq)]
+pub struct Hostname(String);
+impl Hostname {
+	/// Returns the length of the hostname.
+	pub fn len(&self) -> u8 {
```

Comment:
Hmm, I think it is useful to make that invariant explicit though and it allows using guaranteed safe casts instead of having to cast to `u8`. What do you think? I do agree with it conflicting with a common signature though.

Reply:
Yeah, the safe casts are a valid argument. I have no strong opinion either way, fine by me if you leave it `u8`.